internal/clients/team4/client.go

```diff
@@ -200,3 +202,64 @@ func (c *client) VoteForElection(roleToElect shared.Role, candidateList []shared
 
 	return returnList
 }
+
+
+//MonitorIIGORole decides whether to perform monitoring on a role
+//COMPULOSRY: must be implemented
+func (c *client) MonitorIIGORole(roleName shared.Role) bool {
+	
+	presidentID := c.getPresident()
+	speakerID := c.getSpeaker()
+	judgeID := c.getJudge()
+	clientID := shared.ClientID(4)
+	// TODO: Choose sensible thresholds!
+	trustThreshold := 0.5
+	resourcesThreshold := shared.Resources(100)
+	monitoring := false
+	if presidentID == clientID {
+		// If we are the president.
+		monitoring = (c.internalParam.agentsTrust[speakerID] < trustThreshold ||
+					  c.internalParam.agentsTrust[judgeID]   < trustThreshold ) && 
+					 (c.ServerReadHandle.GetGameState().ClientInfo.Resources > resourcesThreshold)		   
+	} else if speakerID == clientID {
+		// If we are the Speaker.
+		monitoring = (c.internalParam.agentsTrust[presidentID] < trustThreshold ||
+					  c.internalParam.agentsTrust[judgeID]   < trustThreshold ) && 
+					 (c.ServerReadHandle.GetGameState().ClientInfo.Resources > resourcesThreshold)
+	} else if judgeID == clientID {
+		// If we are the Judge.
+		monitoring = (c.internalParam.agentsTrust[speakerID] < trustThreshold ||
+					  c.internalParam.agentsTrust[judgeID]   < trustThreshold ) && 
+					 (c.ServerReadHandle.GetGameState().ClientInfo.Resources > resourcesThreshold)
+	}
+	return monitoring
+}
+
+//DecideIIGOMonitoringAnnouncement decides whether to share the result of monitoring a role and what result to share
+//COMPULSORY: must be implemented
+func (c *client) DecideIIGOMonitoringAnnouncement(monitoringResult bool) (resultToShare bool, announce bool) {
+	collaborationThreshold := 0.5
+	importance := mat.NewVecDense(5, []float64{
+		0.0, 1.0, -1.0, 1.0, 0.0,
+	})
+
+	parameters := mat.NewVecDense(5, []float64{
+		c.internalParam.greediness,
+		c.internalParam.selfishness,
+		c.internalParam.fairness,
+		c.internalParam.collaboration,
+		c.internalParam.riskTaking,
+	})
+	// Initialise Return values.
+	announce = true
+	resultToShare = monitoringResult
+
+	// Calculate collaborationLevel based on the current personality of the client.
+	collaborationLevel := mat.Dot(importance, parameters)
+
+	if collaborationLevel > collaborationThreshold {
+		// announce only if we are collaborative enough.
+		announce = true
+	}
+	return resultToShare, announce
```

Comment:
As you're using named returns this isn't strictly necessary.

Reply:
yes I just added to be more explicit, I don't see the harm

Reply:
I believe that if you don't have the return, linter screams at you